# 03 — Ipotesi 2: momentum della vol

BEAR se la vol è in rapido aumento (variazione a 21g nel q80 alto). La dinamica spesso anticipa il livello.

In [1]:
import warnings; warnings.filterwarnings("ignore")
import numpy as np, pandas as pd
import vol_regime as vr
pd.set_option("display.float_format", lambda x: f"{x:.3f}")

vol = vr.load_vol()

## Hit ratio: il segnale predice la direzione del mese successivo? (vs proxy)

In [2]:
for ac, vcol in vr.VOL_FOR_CLASS.items():
    v = vol[vcol].dropna()
    sig = vr.signal_vol_momentum(v, window=21, q=0.80)
    print(f"==== {ac} (vol={vcol}) ====")
    print(vr.compare_table(sig, ac).to_string(float_format=lambda x: f"{x:.3f}"))
    print()

==== Equity (vol=VIX) ====
                hit_ratio       n  base_rate_up  bull_share
Equity (h=21d)                                             
signal              0.546 317.000         0.618       0.776
proxy               0.615 317.000         0.618       0.700

==== Fixed Income (vol=MOVE) ====
                      hit_ratio       n  base_rate_up  bull_share
Fixed Income (h=21d)                                             
signal                    0.513 269.000         0.509       0.803
proxy                     0.510 286.000         0.514       0.493



## Stabilità: hit ratio per anno (Equity)

In [3]:
v = vol["VIX"].dropna()
sig = vr.signal_vol_momentum(v, window=21, q=0.80)
px = vr.load_prices()["SPY"].dropna()
sig_al = sig.reindex(px.index).ffill()
vr.hit_ratio_by_year(sig_al, px).round(3)

2000   0.500
2001   0.417
2002   0.250
2003   0.833
2004   0.333
2005   0.500
2006   0.417
2007   0.250
2008   0.417
2009   0.750
2010   0.583
2011   0.583
2012   0.750
2013   0.667
2014   0.500
2015   0.417
2016   0.583
2017   0.667
2018   0.583
2019   0.583
2020   0.917
2021   0.500
2022   0.500
2023   0.500
2024   0.667
2025   0.583
2026   0.400
Name: correct, dtype: float64

**Conclusione (03).** Il momentum della vol è leggermente più reattivo del
livello ma **non batte il proxy** sull'hit ratio (≈0.55 equity). Vedremo nel 05
che protegge un po' meglio della soglia nei drawdown, ma resta sotto il proxy.